components/shared/SubmitButton.tsx

```diff
@@ -1,75 +1,78 @@
 import { ButtonHTMLAttributes, FC, useMemo } from "react";
 import { css } from "@emotion/react";
 import { Theme } from "@mui/material";
-import { useCENNZWallet } from "@/providers/CENNZWalletProvider";
 import CENNZIconSVG from "@/assets/vectors/cennznet-icon.svg";
 import MetaMaskSVG from "@/assets/vectors/metamask.svg";
 import { useMetaMaskWallet } from "@/providers/MetaMaskWalletProvider";
+import { useWalletProvider } from "@/providers/WalletProvider";
+import { useSelectedAccount } from "@/hooks";
 
 interface SubmitButtonProps {
-	requireCENNZnet: boolean;
-	requireMetaMask: boolean;
+	forceRequireMetaMask?: boolean;
 }
 
 const SubmitButton: FC<
 	ButtonHTMLAttributes<HTMLButtonElement> & SubmitButtonProps
-> = ({ children, requireCENNZnet, requireMetaMask, disabled, ...props }) => {
-	const { selectedAccount: cennzAccount, connectWallet: connectCENNZWallet } =
-		useCENNZWallet();
+> = ({ children, forceRequireMetaMask, disabled, ...props }) => {
+	const { selectedWallet, setWalletOpen } = useWalletProvider();
 	const {
 		selectedAccount: metaMaskAccount,
 		connectWallet: connectMetaMaskWallet,
 	} = useMetaMaskWallet();
+	const selectedAccount = useSelectedAccount();
 
 	const isSubmittable = useMemo(() => {
-		if (requireCENNZnet && !cennzAccount) return false;
-
-		if (requireMetaMask && !metaMaskAccount) return false;
+		if (!selectedAccount) return false;
+		if (forceRequireMetaMask && !metaMaskAccount) return false;
 
 		return true;
-	}, [requireCENNZnet, cennzAccount, requireMetaMask, metaMaskAccount]);
+	}, [forceRequireMetaMask, metaMaskAccount, selectedAccount]);
 
 	return (
 		<>
-			{!cennzAccount && requireCENNZnet && (
+			{!selectedWallet && (
 				<button
 					type="button"
 					css={[styles.root, styles.cennzButton]}
-					onClick={() => connectCENNZWallet()}
+					onClick={() => setWalletOpen(true)}
 				>
 					<img
 						src={CENNZIconSVG.src}
 						alt="CENNZnet Logo"
 						css={styles.brandLogo}
 					/>
-					CONNECT CENNZnet
+					CONNECT WALLET
 				</button>
 			)}
 
-			{!metaMaskAccount && requireMetaMask && !!cennzAccount && (
-				<button
-					type="button"
-					css={[styles.root, styles.metaMaskButton]}
-					onClick={() => connectMetaMaskWallet()}
-				>
-					<img
-						src={MetaMaskSVG.src}
-						alt="MetaMask Logo"
-						css={styles.brandLogo}
-					/>
-					CONNECT METAMASK
-				</button>
-			)}
-
-			{isSubmittable && (
-				<button
-					css={[styles.root, styles.submitButton]}
-					type="submit"
-					disabled={disabled}
-					{...props}
-				>
-					{children}
-				</button>
+			{!!selectedWallet && (
+				<>
+					{!isSubmittable && (
```

Comment:
feel like we are missing a scenario here, when `isSubmittable = false` because no selected account (regardless we need to require MetaMask or not), it will say "CONNECT METAMASK"